pkg/client/vcwallet/client.go

```diff
@@ -9,25 +9,139 @@ package vcwallet
 import (
 	"encoding/json"
 	"fmt"
+	"time"
+
+	"github.com/hyperledger/aries-framework-go/pkg/secretlock"
+	"github.com/hyperledger/aries-framework-go/pkg/storage"
 )
 
 // provider contains dependencies for the verifiable credential wallet client
 // and is typically created by using aries.Context().
 type provider interface {
-	// TODO to be added #2433
+	StorageProvider() storage.Provider
+}
+
+// kmsOpts contains options for creating verifiable credential wallet client.
+type kmsOpts struct {
+	// local kms options
+	secretLockSvc secretlock.Service
+	passphrase    string
+
+	// remote(web) kms options
+	keyServerURL string
+}
+
+// KeyManagerOptions is option for verifiable credential wallet client key manager.
+type KeyManagerOptions func(opts *kmsOpts)
+
+// WithSecretLockService option, when provided then wallet client will use local kms for key operations.
+func WithSecretLockService(svc secretlock.Service) KeyManagerOptions {
+	return func(opts *kmsOpts) {
+		opts.secretLockSvc = svc
+	}
+}
+
+// WithPassphrase option to provide passphrase for local kms for key operations.
+func WithPassphrase(passphrase string) KeyManagerOptions {
+	return func(opts *kmsOpts) {
+		opts.passphrase = passphrase
+	}
+}
+
+// WithKeyServerURL option, when provided then wallet client will use remote kms for key operations.
+// This option will be ignore if provided with 'WithSecretLockService' option.
+func WithKeyServerURL(url string) KeyManagerOptions {
+	return func(opts *kmsOpts) {
+		opts.keyServerURL = url
+	}
 }
 
 // Client enable access to verifiable credential wallet features.
 type Client struct {
 	// ID of wallet content owner
 	userID string
+
+	// wallet profile
+	profile *profile
+
+	// storage provider
+	storeProvider storage.Provider
+}
+
+// Get returns new verifiable credential wallet client for given user.
+// returns error if wallet profile is not found.
+// To create a new client with new wallet profile, use `New()`.
+func Get(userID string, ctx provider) (*Client, error) {
```

Comment:
This is traditionally the New function (new client).

Reply:
fixing it now, will be pushed shortly.

Reply:
fixed